validator_client/src/sync_committee_service.rs

```diff
@@ -336,27 +338,29 @@ impl<T: SlotClock + 'static, E: EthSpec> SyncCommitteeService<T, E> {
             .data;
 
         // Make `SignedContributionAndProof`s
-        let signed_contributions = subnet_aggregators
-            .into_iter()
-            .filter_map(|(aggregator_index, aggregator_pk, selection_proof)| {
-                self.validator_store
-                    .produce_signed_contribution_and_proof(
-                        aggregator_index,
-                        &aggregator_pk,
-                        contribution.clone(),
-                        selection_proof,
+        let mut signed_contributions = Vec::with_capacity(subnet_aggregators.len());
+        for (aggregator_index, aggregator_pk, selection_proof) in subnet_aggregators {
+            match self
+                .validator_store
+                .produce_signed_contribution_and_proof(
+                    aggregator_index,
+                    &aggregator_pk,
+                    contribution.clone(),
+                    selection_proof,
+                )
+                .await
```

Comment:
Similarly there's only one task for signed contributions and proofs, so we should ideally do this in parallel too.